include/hnswlib/space_pq.h

```diff
@@ -0,0 +1,84 @@
+#pragma once
+#include "hnswlib.h"
+#include <memory>
+#include <stdint.h>
+
+namespace hnswlib {
+
+typedef struct pq_dist_param_s {
+  size_t n_subvectors;
+  size_t n_clusters;
+  float *dist_mat;
+} pq_dist_param_t;
+
+template <typename CODETYPE>
+static float PQLookup(const void *pVect1v, const void *pVect2v,
```

Comment:
why don't u do `const CODETYPE* pVect1v` and let the compiler capture errors?

Reply:
That's a interface from `hnswlib`
https://github.com/jina-ai/annlite/blob/103b5bf31b59ca041d7ca70b8e0442421bcdf279/include/hnswlib/hnswlib.h#L48-L49